src/lib/proxy/global-frontend-api-cache.ts

```diff
@@ -88,6 +90,13 @@ export class GlobalFrontendApiRepository extends EventEmitter {
         try {
             this.featuresByEnvironment = await this.getAllFeatures();
             this.segments = await this.getAllSegments();
+            if (this.status === 'starting') {
```

Comment:
cache is event emitter itself and events make it easy to write tests without artificial wait times